### ISSUE TITLE
Efficient Net UnavailableError Bad hardware status when training b5-b7

### ISSUE THREAD

**Issue by adfi** [tensorflow/tpu#403]
Hi, I've been using the EfficientNet code to train on my own images and that has been going well for variants b0 to b4. However when I select variants b5 to b7 I keep getting the following error:

> Caused by op 'input_pipeline_task0/while/InfeedQueue/enqueue/1', defined at:
  File "main.py", line 725, in <module>
    app.run(main)
  File "/usr/local/lib/python3.5/dist-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.5/dist-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "main.py", line 697, in main
    est.train(input_fn=imagenet_train.input_fn, max_steps=next_checkpoint)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2452, in train
    saving_listeners=saving_listeners)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 358, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1124, in _train_model
    return self._train_model_default(input_fn, hooks, saving_listeners)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1154, in _train_model_default
    features, labels, model_fn_lib.ModeKeys.TRAIN, self.config)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2251, in _call_model_fn
    config)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1112, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2547, in _model_fn
    input_holders.generate_infeed_enqueue_ops_and_dequeue_fn())
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 1167, in generate_infeed_enqueue_ops_and_dequeue_fn
    self._invoke_input_fn_and_record_structure())
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 1271, in _invoke_input_fn_and_record_structure
    wrap_fn(device=host_device, op_fn=enqueue_ops_fn))
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2945, in _wrap_computation_in_while_loop
    parallel_iterations=1)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/control_flow_ops.py", line 3556, in while_loop
    return_same_structure)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/control_flow_ops.py", line 3087, in BuildLoop
    pred, body, original_loop_vars, loop_vars, shape_invariants)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/control_flow_ops.py", line 3022, in _BuildLoop
    body_result = body(*packed_vars_for_body)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2934, in computation
    with ops.control_dependencies(op_fn()):
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 894, in enqueue_ops_fn
    tpu_ordinal_function=tpu_ordinal_function_impl)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_feed.py", line 526, in generate_enqueue_ops
    for (shard, index) in zip(sharded_inputs, xrange(self.number_of_shards))
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_feed.py", line 526, in <listcomp>
    for (shard, index) in zip(sharded_inputs, xrange(self.number_of_shards))
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_feed.py", line 458, in _generate_enqueue_op
    device_ordinal=tpu_ordinal)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/ops/gen_tpu_ops.py", line 1314, in infeed_enqueue_tuple
    device_ordinal=device_ordinal, name=name)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3300, in create_op
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1801, in __init__
    self._traceback = tf_stack.extract_stack()
UnavailableError (see above for traceback): Bad hardware status: 0x1
         [[node input_pipeline_task0/while/InfeedQueue/enqueue/1 (defined at /usr/local/lib/python3.5/dist-packages/tensorflow_estimator/python/estimator/estimator.py:1112) ]]

I ran the following command for training:
`python3 main.py --tpu=${TPU_NAME} --data_dir=${DATA_DIR} --model_dir=${STORAGE_BUCKET}/efficientnet-b6 --model_name='efficientnet-b6' --skip_host_call=true --train_batch_size=2048  --train_steps=10000 --num_eval_images=9000 --num_label_classes=2019 --steps_per_eval=1000`

At first I thought it was an issue with the input data so I played around with the batch size and train steps but that didn't help. I even changed the input data to a different data set but that didn't help either.

I followed the EfficientNet [tutorial](https://cloud.google.com/tpu/docs/tutorials/efficientnet) to set up the TPU.

**Comment by saberkun** [comment]
Hi @adfi, This is very likely from OOM. When you use b5-b7 model, the model is much larger and cost more memory. If you keep the same batch size, you will not be able to fit into TPU memory.
Please consider --train_batch_size=1024 or even smaller.
"Bad hardware status" is not the 'true' error message.

**Comment by adfi** [comment]
@saberkun thanks for the reply. I did change the train batch size but turns out I also needed to the change the eval batch size.
